pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -186,26 +189,48 @@ public String listMetadataForOneSegment(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
       @ApiParam (value = "Name of segment", required = true) @PathParam("segmentName") @Encoded String segmentName,
       @ApiParam(value = "realtime|offline", required = false) @QueryParam("type") String tableTypeStr
-  ) {
+  ) throws JSONException {
     segmentName = checkGetEncodedParam(segmentName);
     CommonConstants.Helix.TableType tableType = Constants.validateTableType(tableTypeStr);
+    return listSegmentMetadataInternal(tableName, segmentName, tableType);
+  }
+
+  private String listSegmentMetadataInternal(@Nonnull  String tableName, @Nonnull String segmentName,
+      @Nullable CommonConstants.Helix.TableType tableType) throws JSONException {
     // The code in restlet.resources seems to return an array of arrays, so we will do the same
     // to maintain backward compatibility
     JSONArray result = new JSONArray();
```

Comment:
yes, the two APIs now function same, except one returns a json array of strings, and the other returns a json array of objects.